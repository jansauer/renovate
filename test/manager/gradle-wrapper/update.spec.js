const fs = require('fs');
const dcUpdate = require('../../../lib/manager/gradle-wrapper/update');

const propertiesFile1 = fs.readFileSync(
  'test/_fixtures/gradle-wrapper/gradle-wrapper-1.properties',
  'utf8'
);
const propertiesFile2 = fs.readFileSync(
  'test/_fixtures/gradle-wrapper/gradle-wrapper-2.properties',
  'utf8'
);

describe('manager/gradle-wrapper/update', () => {
  describe('updateDependency', () => {
    it('replaces existing value', () => {
      const upgrade = {
        toVersion: '5.0.0',
        lineNumber: 5,
        shaLineNumber: 6,
        version: '5.0.0',
        downloadUrl:
          'https://services.gradle.org/distributions/gradle-5.0-all.zip',
        checksum:
          '336b6898b491f6334502d8074a6b8c2d73ed83b92123106bd4bf837f04111043',
        digests: 'true',
      };
      const res = dcUpdate.updateDependency(propertiesFile2, upgrade);
      expect(res).toMatchSnapshot();
      expect(res).not.toEqual(propertiesFile2);
      expect(res.includes(upgrade.downloadUrl.replace(':', '\\:'))).toBe(true);
      expect(res.includes(upgrade.checksum)).toBe(true);
    });

    it('replaces existing value and add digest', () => {
      const upgrade = {
        toVersion: '5.0.0',
        lineNumber: 4,
        version: '5.0.0',
        downloadUrl:
          'https://services.gradle.org/distributions/gradle-5.0-all.zip',
        checksum:
          '336b6898b491f6334502d8074a6b8c2d73ed83b92123106bd4bf837f04111043',
        digests: 'true',
      };
      const res = dcUpdate.updateDependency(propertiesFile1, upgrade);
      expect(res).toMatchSnapshot();
      expect(res).not.toEqual(propertiesFile1);
      expect(res.includes(upgrade.downloadUrl.replace(':', '\\:'))).toBe(true);
      expect(res.includes(upgrade.checksum)).toBe(true);
    });

    it('replaces existing value and remove digest', () => {
      const upgrade = {
        toVersion: '4.10.3',
        lineNumber: 5,
        shaLineNumber: 6,
        version: '4.10.3',
        downloadUrl:
          'https://services.gradle.org/distributions/gradle-4.10.3-all.zip',
        digests: 'false',
      };
      const res = dcUpdate.updateDependency(propertiesFile2, upgrade);
      expect(res).toMatchSnapshot();
      expect(res.includes(upgrade.downloadUrl.replace(':', '\\:'))).toBe(true);
    });

    it('returns same', () => {
      const upgrade = {
        toVersion: '4.10.3',
        lineNumber: 5,
        shaLineNumber: 6,
        version: '4.10.3',
        downloadUrl:
          'https://services.gradle.org/distributions/gradle-4.10.3-all.zip',
        checksum:
          '336b6898b491f6334502d8074a6b8c2d73ed83b92123106bd4bf837f04111043',
        digests: 'true',
      };
      const res = dcUpdate.updateDependency(propertiesFile2, upgrade);
      expect(res).toEqual(propertiesFile2);
    });

    it('returns null if error', () => {
      const res = dcUpdate.updateDependency(null, null);
      expect(res).toBe(null);
    });
  });
});
