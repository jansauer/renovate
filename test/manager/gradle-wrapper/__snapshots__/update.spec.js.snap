// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager/gradle-wrapper/update updateDependency replaces existing value 1`] = `
"distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
# @See https://gradle.org/releases/
distributionUrl=https\\\\://services.gradle.org/distributions/gradle-5.0-all.zip
distributionSha256Sum=17847c8e12b2bcfce26a79f425f082c31d4ded822f99a66127eee2d96bf18216
"
`;

exports[`manager/gradle-wrapper/update updateDependency replaces existing value and add digest 1`] = `
"distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\\\\://services.gradle.org/distributions/gradle-5.0-bin.zip
distributionSha256Sum=6157ac9f3410bc63644625b3b3e9e96c963afd7910ae0697792db57813ee79a6"
`;
